packages/bot-web-ui/src/pages/bot-builder/quick-strategy/selects/growth-rate-type.tsx

```diff
@@ -0,0 +1,190 @@
+                setFieldError('tick_count', min_error);
+                prev_error.current.tick_count = min_error;
+            } else {
+                prev_error.current.tick_count = null;
+                setFieldError('tick_count', undefined);
+            }
+            prev_error.current.take_profit = null;
+        } catch (error_response) {
+            setFieldError('take_profit', error_response?.error?.message);
+            prev_error.current.take_profit = error_response?.error?.message;
+        }
+    };
+
+    const debounceChange = React.useCallback(
+        debounce(
+            values => {
+                validateMinMaxForAccumulators(values);
+            },
+            500,
+            {
+                trailing: true,
+                leading: false,
+            }
+        ),
+        []
+    );
+
+    React.useEffect(() => {
+        if (
+            prev_proposal_payload.current?.symbol !== values.symbol ||
+            prev_proposal_payload.current?.amount !== values.amount ||
+            prev_proposal_payload.current?.limit_order?.take_profit !== values.take_profit ||
+            prev_proposal_payload.current?.currency !== client.currency ||
+            prev_proposal_payload.current?.growth_rate !== values.growth_rate
+        ) {
+            debounceChange(values);
+        }
+    }, [values.take_profit, values.tick_count, values.amount, currency]);
+
+    const handleChange = async (value: string) => {
+        setFieldValue?.(name, value);
+        setValue(name, value);
+    };
+
+    const key = `qs-contract-type-${name}`;
+
+    return (
+        <div className='qs__form__field qs__form__field__input no-top-spacing'>
+            <Field name={name} key={key} id={key}>
+                {({ field }: FieldProps) => {
+                    const selected_item = list?.find(item => item?.value === field?.value);
+                    if (!is_desktop) {
+                        return (
+                            <ul className='qs__form__field__list' data-testid='dt_qs_contract_types'>
+                                {list.map(item => {
+                                    const is_active = selected_item?.value === item?.value;
+                                    return (
+                                        <li
+                                            key={item?.value}
+                                            className={classNames('qs__form__field__list__item', {
+                                                'qs__form__field__list__item--active': is_active,
+                                            })}
+                                            onClick={() => {
+                                                handleChange(item?.value);
+                                            }}
+                                            onChange={() => {
+                                                handleChange(item?.value);
+                                            }}
```

Comment:
Do we need both or only `onChange` should work?

Reply:
@rupato-deriv do we need both `onClick` and `onChange` here?

Reply:
onChange is needed done ✅